torch/distributed/pipelining/schedules.py

```diff
@@ -2160,6 +2161,165 @@ def need_bubble(stage, op, microbatch, num_stages_global, seen_ops):
         return result
 
 
+class ScheduleZBVZeroBubble(PipelineScheduleMulti):
+    """
+    The Zero Bubble schedule (ZBV variant).
+    See https://arxiv.org/pdf/2401.10241 Section 6 for details.
+    This schedule wille perform one forward and one backward on inputs for the microbatches in steady
+    state and supports multiple stages per rank. Uses the backward for weights to fill in
```

Comment:
does this schedule actually require 2 stages per rank? or does it support more than 2 stages per rank

Reply:
it would also be good to mention that this ZB-V schedule would have the "zero bubble" property only if tF=tI=tW, which is likely not strictly true for real models.  Alternative greedy scheduler could be implemented for unequal/unbalanced time